core/src/main/java/org/kohsuke/stapler/export/Property.java

```diff
@@ -199,118 +197,122 @@ private void writeValue(Type expected, Object value, TreePruner pruner, DataWrit
+        }
+        Class act = c.getComponentType();
+        if (act !=null) { // array
+            Range r = pruner.getRange();
+            writer.startArray();
+            if (value instanceof Object[]) {
+                // typical case
+                for (Object item : r.apply((Object[]) value)) {
+                    writeBuffered(act, item, pruner, writer);
                 }
-                writer.endArray();
-                return;
-            }
-            if(value instanceof Iterable) {
-                writer.startArray();
-                Type expectedItemType = Types.getTypeArgument(expected, 0, null);
-                for (Object item : pruner.getRange().apply((Iterable) value)) {
-                    writeBuffered(expectedItemType, item, pruner, writer);
+            } else {
+                // more generic case
+                int len = Math.min(r.max, Array.getLength(value));
+                for (int i=r.min; i<len; i++) {
+                    writeBuffered(act, Array.get(value, i), pruner, writer);
                 }
-                writer.endArray();
-                return;
             }
-            if(value instanceof Map) {
-                if (verboseMap!=null) {// verbose form
-                    writer.startArray();
-                    for (Map.Entry e : ((Map<?,?>) value).entrySet()) {
-                        BufferedDataWriter buffer = new BufferedDataWriter(writer.getExportConfig());
-                        try {
-                            writeStartObjectNullType(buffer);
-                            buffer.name(verboseMap[0]);
-                            writeValue(null, e.getKey(), pruner, buffer);
-                            buffer.name(verboseMap[1]);
-                            writeValue(null, e.getValue(), pruner, buffer);
-                            buffer.endObject();
-                            buffer.finished();
-                        } catch (IOException x) {
-                            if (x.getCause() instanceof InvocationTargetException) {
-                                LOGGER.log(Level.WARNING, "skipping export of " + e, x);
-                            }
+            writer.endArray();
+            return;
+        }
+        if(value instanceof Iterable) {
+            writer.startArray();
+            Type expectedItemType = Types.getTypeArgument(expected, 0, null);
+            for (Object item : pruner.getRange().apply((Iterable) value)) {
+                writeBuffered(expectedItemType, item, pruner, writer);
+            }
+            writer.endArray();
+            return;
+        }
+        if(value instanceof Map) {
+            if (verboseMap!=null) {// verbose form
+                writer.startArray();
+                for (Map.Entry e : ((Map<?,?>) value).entrySet()) {
+                    BufferedDataWriter buffer = new BufferedDataWriter(writer.getExportConfig());
+                    try {
+                        writeStartObjectNullType(buffer);
```

Comment:
Actually the [whitespace-reduced diff](https://github.com/stapler/stapler/pull/127/files?w=1) is more approachable; there must have been a change in indentation for a long block of code.